### PR TITLE
[Hub Generated] Review request for Microsoft.CostManagement to add version stable/2019-11-01

### PR DESCRIPTION
**Pull request by sashahdo**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/sashahdo/azure-rest-api-specs/dev-cost-management-Microsoft.CostManagement-2019-11-01...master/).

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json [hunk 1]
```diff
@@ -317,6 +317,103 @@
...
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "patch": {
```
Comment (allenjzhang) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json:364]: Mmm, not sure about using Patch as an action (not just for updating state to signal dismiss state). This is typically done with POST. Calling it out for ARM review.

Comment (sashahdo) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json:364]: thanks for pointing out the enum issue. I made an update adding all missing enums.

Comment (majastrz) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json:364]: I'm really confused about the purpose of the API. The contract looks like a legitimate PATCH operation, but you call it a "dismiss" API.

If you are planning to allow patching of other alert properties in the future, then I think it makes sense to keep PATCH as the verb but you should change all the descriptions. ("Dismiss" in examples is fine, though.)

If dismissing is the only modification that you will allow, then it should be a POST /dismiss on the alert resource.

---
In reply to: [421001556](https://github.com/Azure/azure-rest-api-specs/pull/9329#discussion_r421001556) [](ancestors = 421001556)

Comment (theelderwand) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json:364]: @majastrz  That's great feedback. Completely agree with you on this one. It should be a /dismiss. However, this is the currently published API which customers are actively using so we are not looking to make any breaking changes to contract here. When we do rev the API version in the near future we will incorporate this.
Do let me know if you see any concerns on this.

Comment (majastrz) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2019-11-01/costmanagement.json:364]: It's fine to keep it, but you should update the description accordingly (except for examples). That change is swagger-only and has no bearing on your backend.

**Comment by sashahdo** [comment]
rebased with upstream master to resolve merge conflicts

**Comment by allenjzhang** [comment]
@RodgeFu , Can you take a look CLI automation failure? Thanks!

**Comment by sashahdo** [comment]
this pr has gotten quite ugly so I'm going to go ahead and abandon it and create a new one.

**Pull request closed without merging**